prefix_metrics_test.go

```diff
@@ -0,0 +1,38 @@
+package gometer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrefixMetricsRegister(t *testing.T) {
+	originalMetrics := New()
+	prefixMetrics := NewPrefixMetrics(originalMetrics, "data.")
+
+	counter1 := &Counter{}
+	counter1.Set(11)
+	counter2 := &Counter{}
+	counter2.Set(22)
+
+	require.Nil(t, prefixMetrics.Register("first", counter1))
+	require.Nil(t, prefixMetrics.Register("second", counter2))
+
+	expected := map[string]*Counter{
+		"data.first":  counter1,
+		"data.second": counter2,
+	}
+	assert.Equal(t, expected, originalMetrics.counters)
```

Comment:
I'd suggest not to use private fields in tests because we need to test a public API. After you've registered the metrics you can simply get it or write it somewhere.

Reply:
fixed